gips/0083.md

```diff
@@ -0,0 +1,240 @@
+
+# Specification
+
+## Permissionless Discovery
+
+This section discusses how Indexers may be listed on the Payments Gateway UI.
+
+Substreams providers MUST register their service on-chain by publishing a _Substreams Service Deployment_ manifest to IPFS, and [`allocate`ing](https://github.com/graphprotocol/contracts/blob/main/packages/contracts/contracts/staking/Staking.sol#L296) to it through the Staking contract on the Arbitrum chain.
+
+Here is a sample _Substreams Service Deployment_ manifest:
+```yaml
+specVersion: 0.0.5
+description: "Substreams Data Service for MyNetwork"
+service:
+  type: substreams-v1
+  endpoint: https://mynetwork.substreams.example.com
+  network: mynetwork
+  provider:
+    id: my-company
+    name: My Company
+    logo: https://company.example.com/logo.png
+```
+
+The service `substreams-v1` means that the endpoint makes available a gRPC endpoint responding to these methods:
+
+- sf.substreams.rpc.v2.Stream/Block
+- sf.substreams.rpc.v2.EndpointInfo/Info
+
+as specified by https://github.com/streamingfast/substreams/blob/develop/proto/sf/substreams/rpc/v2/service.proto updated from time-to-time in backwards compatible ways.
+
+The Payment Gateway MUST listen to the Arbitrum Staking contract for such registrations (through a Substreams module streamed, a Subgraph like the canonical Network Subgraph or other means), and update its local view of the network.
+
+The Payment Gateway CAN have systems to perform health checks (e.g. on `/health` or `/info` endpoints), or other checks to ensure active, up-to-date and properly configured providers are offered to end users. A Payment Gateway CAN check block height to ensure the backing provider is close to chain head before offering it to users.
+
+
+## Indexer Selection Algorithm (ISA)
+
+To provide consumers with fair access to Substreams providers, an Indexer Selection Algorithm is needed.
+
+Given that there's no gateway involved with Substreams (it's direct point-to-point between consumer and provider), the existing Indexer Selection Algorithm (ISA), which happens at the time of query, cannot be applied for Substreams.  Therefore, a round-robin selection of providers is offered on the front-end of the Payment Gateway (https://thegraph.market), ensuring fair distribution of request loads, while still allowing consumers to choose their provider.
+
+> NON-NORMATIVE: A reason for selecting a provider could be that it has pre-cached data for a given dataset, which would speed up and lower the costs for a consumer.
+
+> NON-NORMATIVE: Future GIPs will propose improved ISA, based on better data and better metrics, which might be desirable to consumers.
+
+## Payments
+
+Payments are handled by the Payment Gateway, which is responsible for collecting payments from consumers and distributing them to indexers.
```

Comment:
As per [TG's style guidelines](https://drive.google.com/file/d/13tTWe58kKLSDzhNNNTouyktJaDnEEEr-/view), "indexers" should be capitalized as "Indexers."